industrialucn/optimization/econstraint.py

```diff
@@ -1,62 +1,125 @@
 import logging
+from typing import Union, List, Dict, Callable, NoReturn
 
-from docplex.mp.model import Model
+import numpy as np
+from docplex.mp.linear import LinearExpr
+from docplex.mp.model import Model as CplexModel
+from gurobipy import GRB, quicksum, LinExpr
+from gurobipy import Model as GurobiModel
```

Comment:
Add a try/catch here.
Not everyone has gurobi AND cplex installed in their machines.

Reply:
Each optimizer will raise an error and will terminate the script if the user does not have the corresponding installation, which is the desired behavior. I don't see a need for catching and handling it as an exception.